apps/dotcom/src/components/IFrameProtector.tsx

```diff
@@ -76,24 +89,28 @@ export function IFrameProtector({
 
 			if (event.data === EXPECTED_RESPONSE) {
 				// todo: check the origin?
-				setEmbeddedState('iframe-ok')
+				setEmbeddedState(EMBEDDED_STATE.IFRAME_OK)
 				clearTimeout(timeout)
 			}
 		}
 
 		window.addEventListener('message', handleMessageEvent, false)
 
-		if (embeddedState === 'iframe-unknown') {
+		if (embeddedState === EMBEDDED_STATE.IFRAME_UNKNOWN) {
 			// We iframe embeddings on multiplayer or readonly
 			if (WHITELIST_CONTEXT.includes(context)) {
 				window.parent.postMessage(EXPECTED_QUESTION, '*') // todo: send to a specific origin?
```

Comment:
feel free to call it out of scope of this diff, but how necessary is all this message-passing stuff at this stage? I feel like we could probably get by with just checking against the origin - i experimented with this here: https://github.com/tldraw/tldraw/pull/3214/files#diff-10fd47d203ff90d59bdb8ff3675076dfbb01942674b58450f529d2754b592554L58-L103

Reply:
@mimecuvalo we discussed allowing embedding of iFrames - just not creating new rooms from within them. Feels like we could completely remove these check? We'd still keep the `WHITELIST_CONTEXT` since we want to prevent things like local rooms from working in iframes?

Reply:
I would agree that it's overkill. I think it's out-of-the-scope of this particular PR to refactor it though. (the iframe change here was just to add tracking, and I incidentally just made some constants).

Let's take a look in a separate PR to rework it!